app/cdap/components/PostRunActions/PostRunActionsWizard.js

```diff
@@ -99,7 +146,105 @@ export default class PostRunActionsWizard extends Component {
+              <h4 className="modal-title pull-left">Alerts</h4>
+            )}
+            {this.state.errorCount === 0 && <span className="text-success">No errors found.</span>}
+            {this.state.errorCount === 1 && <span className="text-danger">1 error found</span>}
+            {this.state.errorCount > 1 && (
+              <span className="text-danger">{this.state.errorCount} errors found</span>
+            )}
+            <div className="btn-group pull-right">
+              {this.state.showValidateButton && (
+                <button
+                  className="btn btn-primary validate-btn"
+                  data-cy="validate-btn"
+                  type="button"
+                  onClick={this.validatePluginProperties.bind(this)}
+                >
+                  {this.state.isValidating ? (
+                    <span>
+                      <span className="fa fa-spinner fa-spin"></span>
+                    </span>
+                  ) : (
+                    <span>Validate</span>
+                  )}
+                </button>
+              )}
+              <a className="btn" onClick={this.toggleAndPreventPropagation}>
+                <IconSVG name="icon-close" />
+              </a>
+            </div>
+          </div>
+          <ModalBody>
+            {action && this.props.mode === 'edit' ? (
+              <WizardNav
+                mode={this.props.mode}
+                action={this.props.action}
+                currentStage={currentStage}
+                artifact={this.props.artifact}
+                setAction={this.setActionHandler.bind(this)}
+                actionCreator={this.props.actionCreator}
+                toggle={this.toggleAndPreventPropagation}
+                getPostActions={this.props.getPostActions}
+                setShowValidateButton={this.setShowValidateButton.bind(this)}
+                propertyErrors={this.state.propertyErrors}
+                validatePluginProperties={this.props.validatePluginProperties}
+                setState={this.setState.bind(this)}
+              />
+            ) : null}
+            {action && this.props.mode === 'view' ? (
+              <div className="postrun-configuration-content">{this.renderBody()}</div>
+            ) : null}
+            {this.props.mode === 'create' ? (
+              <WizardNav
+                mode={this.props.mode}
+                currentStage={currentStage}
+                artifact={this.props.artifact}
+                setAction={this.setActionHandler.bind(this)}
+                actionCreator={this.props.actionCreator}
+                toggle={this.toggleAndPreventPropagation}
+                getPostActions={this.props.getPostActions}
+                setShowValidateButton={this.setShowValidateButton.bind(this)}
+                propertyErrors={this.state.propertyErrors}
+                validatePluginProperties={this.props.validatePluginProperties}
+                setState={this.setState.bind(this)}
```

Comment:
i don't think its a good idea to pass this.setState down into a child component - please break out what you were going to set into a separate function and pass that down. Is this component a copy of the one above?